### DISCUSSION TITLE
Camera Support

### DISCUSSION THREAD

**Discussion by Anton-Gasse** [pygame-web/pygbag#147] [Q&A] [answered]
Hello,
I was wondering if it is possible to use the pygbag module on a python script that uses the camera in a pygame Script. I have tested it with the opencv VideoCapture and the pygame camera module and with both ways it was working on my local maschine as a normal python script, but when i compiled it with pygbag and and tried it in my browser it wasn't working anymore.
I think it's because finding the camera works different when starting a python script on your maschine then the compiled version in the browser. But maybe there is already a workaround. If not this would be an awesome new feature ;)
Kind regards Anton

**Comment by pmp-p** [comment, marked as answer]
it is possible, just need async and SSL ( or localhost pygbag )

[test](https://pygame-web.github.io/showroom/pythongit.html#src/test_vidcap.py)
code https://github.com/pygame-web/showroom/blob/main/src/test_vidcap.py

the thin async layer module is also compatible with desktop and is here https://github.com/pygame-web/platform_wasm/blob/main/platform_wasm/pygame/vidcap.py

**Reply by Anton-Gasse** [reply]
Hey, thanks for the quick answer. I am pretty impressed that it is actually possible to access the camera. Unfortunately, I am not quite sure how I could run the code you sent me, on my device. I would be really grateful for a short explanation.

**Reply by pmp-p** [reply]
The camera module makes use of pygame-ce.
Regarding pygame-ce specific help you'll get better results on [pygame discord](https://discord.com/invite/pygame) help sub channels and pygame-web

**Reply by Anton-Gasse** [reply]
Okay interesting, but i would like to know how to run your scripts that you send me. I would be really happy with a short explanation on how to run these two files locally on my maschine and if there is something i have to keep an eye on when compiling it with the pygbag prompt.

**Reply by pmp-p** [reply]
maybe copy the content of test_vidcap.py in/as your main.py and elaborate from that ? make sure you test tested first with the [test] link in first reply, webcam must be plugged in before loading page.

**Reply by Anton-Gasse** [reply]
Yes i have tried this but it did not work. I think my Problem is that it uses the second script and i am not quiet sure how to combine them.

**Reply by pmp-p** [reply]
the second script "vidcap.py" is the implementation of camera async module itself, it is already in pygbag you don't need it when in browser
